tests/mechanic/supplier_test.py

```diff
@@ -353,15 +353,19 @@ def docker_log_output(text):
     def test_check_container_return_code(self, caplog):
         mock_docker_client = mock.MagicMock()
         builder = supplier.DockerBuilder(src_dir="/src", build_jdk=8, log_dir="logs", client=mock_docker_client)
-
         completion = {"StatusCode": 0, "Error": None}
         builder.check_container_return_code(completion, "test-container-name")
         assert "Container [test-container-name] completed successfully." in caplog.text
 
         completion = {"StatusCode": 1, "Error": "Vague error message"}
-        with pytest.raises(exceptions.SystemSetupError) as e:
-            builder.check_container_return_code(completion, "test-container-name")
-        assert "Docker container [test-container-name] failed with status code [1]: Error [Vague error message]" in str(e.value)
+        with mock.patch("builtins.open", mock.mock_open()) as file:
+            file.return_value.readlines.return_value = "my error"
```

Comment:
Which is this needed, give that the [check](https://github.com/elastic/rally/pull/1629/files#diff-bcbcac752601b8c90ba4cc053a13169690cea0da315ea09ced558fde60727adcR867) is `if completion["StatusCode"] != 0`?

Reply:
Leftover from my initial approach to regex for the full error message (but ran into complications with escaping characters). Removed in https://github.com/elastic/rally/pull/1629/commits/754598d65fd1be731787c58467d8f8234a7fba84